src/ott/tools/plot.py

```diff
@@ -245,3 +260,159 @@ def animate(
         interval=1000 / frame_rate,
         blit=True
     )
+
+
+# TODO(zoepiran): add support for data of d > 2 (PCA on all k's)
+class PlotMM(Plot):
+  """Plot an optimal transport map for MM-Sinkhorn.
+
+  It enables to either plot or update a plot in a single object, offering the
+  possibilities to create animations as a
+  :class:`~matplotlib.animation.FuncAnimation`, which can in turned be saved to
+  disk at will. There are two design principles here:
+
+  #. we do not rely on saving to/loading from disk to create animations
+  #. we try as much as possible to disentangle the transport problem from
+       its visualization.
+
+  Args:
+    fig: Specify figure object. Created by default
+    ax: Specify axes objects. Created by default
+    threshold: value below which links in transportation matrix won't be
+      plotted. This value should be negative when using animations.
+    cmap: color map used to plot line colors.
+    scale_alpha_by_coupling: use or not the coupling's value as proxy for alpha
+    alpha: default alpha value for lines.
+    title: title of the plot.
+  """
+
+  def __init__(
+      self,
+      fig: Optional["plt.Figure"] = None,
+      ax: Optional["plt.Axes"] = None,
+      cmap: str = "cividis_r",
+      scale_alpha_by_coupling: bool = False,
+      alpha: float = 0.7,
+      title: Optional[str] = None
+  ):
+
+    super().__init__(
+        fig=fig,
+        ax=ax,
+        cmap=cmap,
+        scale_alpha_by_coupling=scale_alpha_by_coupling,
+        alpha=alpha,
+        title=title
+    )
+
+    self._patches = []
+    self._points = []
+    self._n, self._k, self._top_k = None, None, None
+
+  def __call__(self, ot: mmsinkhorn.MMSinkhornOutput) -> List["plt.Artist"]:
+    """Plot 2-D couplings. does not support higher dimensional."""
+    n_s = [len(ot.x_s[i]) for i in range(len(ot.x_s))]
+    assert self._n < jnp.prod(
+        jnp.array(n_s)
+    ), "Intended number of tuples too large."
+
+    # Extract top_k largest entries in the tensor, and their indices.
+    val, idx = jax.lax.top_k(ot.tensor.ravel(), self._top_k)
```

Comment:
I'd consider passing `top_k` instead of saving it. Then in the notebook, you can use the `__call__` method directly for static plotting as well, instead of re-defining the functions.